packages/backend-output-storage/src/stack_metadata_output_storage_strategy.ts

```diff
@@ -34,4 +40,59 @@ export class StackMetadataBackendOutputStorageStrategy
       stackOutputs: Object.keys(backendOutputEntry.payload),
     });
   };
+
+  /**
+   * Lazily construct and append to output list as stack output and add metadata to the metadata object.
+   */
+  appendToBackendOutputList = (
+    keyName: string,
+    backendOutputEntry: BackendOutputEntry
+  ): void => {
+    const version = backendOutputEntry.version;
+
+    Object.entries(backendOutputEntry.payload).forEach(([listName, value]) => {
+      // prevent prototype-polluting assignment
+      if (
+        keyName === '__proto__' ||
+        keyName === 'constructor' ||
+        keyName === 'prototype'
+      ) {
+        return;
+      }
+      if (this.lazyLists[keyName]?.[listName]) {
+        this.lazyLists[keyName][listName].push(value);
+      } else {
+        const outputList: string[] = [value];
+        if (this.lazyLists[keyName]) {
+          this.lazyLists[keyName][listName] = outputList;
```

Comment:
might be easier to read / require less defensive logic if we use Maps instead of objects for this

Reply:
Made the change, looks cleaner compared to `[keyName][listName]` IMO